sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/JdbcUtils.java

```diff
@@ -0,0 +1,119 @@
+        RS_TYPE_MAPPING.put(Types.CHAR, "CHAR");
+        RS_TYPE_MAPPING.put(Types.NUMERIC, "NUMERIC");
+        RS_TYPE_MAPPING.put(Types.DECIMAL, "DECIMAL");
+        RS_TYPE_MAPPING.put(Types.INTEGER, "INTEGER");
+        RS_TYPE_MAPPING.put(Types.SMALLINT, "SMALLINT");
+        RS_TYPE_MAPPING.put(Types.FLOAT, "FLOAT");
+        RS_TYPE_MAPPING.put(Types.REAL, "REAL");
+        RS_TYPE_MAPPING.put(Types.DOUBLE, "DOUBLE");
+        RS_TYPE_MAPPING.put(Types.VARCHAR, "VARCHAR");
+        RS_TYPE_MAPPING.put(Types.DATE, "DATE");
+        RS_TYPE_MAPPING.put(Types.TIME, "TIME");
+        RS_TYPE_MAPPING.put(Types.TIMESTAMP, "TIMESTAMP");
+        RS_TYPE_MAPPING.put(Types.OTHER, "OTHER");
+    }
+
+    private static Schema.FieldType fromFieldType(String typeName) { //List<TableFieldSchema> nestedFields
+        switch (typeName) {
+            case "VARCHAR":
+            case "CHAR":
+            case "LONGVARCHAR":
+            case "CLOB":
+                return Schema.FieldType.STRING;
+            case "BIT":
+                return Schema.FieldType.BOOLEAN;
+            case "NUMERIC":
+                return Schema.FieldType.INT64;
+            case "TINYINT":
+                return Schema.FieldType.BYTE;
+            case "SMALLINT":
+                return Schema.FieldType.INT16;
+            case "INTEGER":
+                return Schema.FieldType.INT32;
+            case "BIGINT":
+                return Schema.FieldType.INT64;
+            case "REAL":
+            case "FLOAT":
+                return Schema.FieldType.FLOAT;
+            case "DOUBLE":
+                return Schema.FieldType.DOUBLE;
+            case "VARBINARY":
+            case "BINARY":
+            case "BLOB":
+                return Schema.FieldType.BYTES;
+            case "DATE":
+                return Schema.FieldType.logicalType(
+                        new LogicalTypes.PassThroughLogicalType<Instant>(
+                                "SqlDateType", "", Schema.FieldType.DATETIME) {});
+            case "TIME":
+                return Schema.FieldType.logicalType(
+                        new LogicalTypes.PassThroughLogicalType<Instant>(
+                                "SqlTimeType", "", Schema.FieldType.DATETIME) {});
+            case "TIMESTAMP":
+                return Schema.FieldType.logicalType(
+                        new LogicalTypes.PassThroughLogicalType<Instant>(
+                                "SqlTimestampWithLocalTzType", "", Schema.FieldType.DATETIME) {});
+            case "DATETIME":
+                return Schema.FieldType.DATETIME;
+//            case "REF":
+//                return Schema.FieldType.?
+//            case "ARRAY":
+//                return Schema.FieldType.?
+
+//            case "STRUCT":
+//            case "RECORD":
+//                Schema rowSchema = fromTableFieldSchema(nestedFields);
+//                return Schema.FieldType.row(rowSchema);
+            default:
```

Comment:
Since there are several JDBC types that don't have a corresponding Beam type, the default action should be to throw an exception.

Reply:
Done